libs/langgraph/src/func.ts

```diff
@@ -0,0 +1,114 @@
+import { BaseCheckpointSaver } from "@langchain/langgraph-checkpoint";
+import { Pregel } from "./pregel/index.js";
+import { PregelNode } from "./pregel/read.js";
+import { END, START } from "./graph/graph.js";
+import { BaseChannel } from "./channels/base.js";
+import { ChannelWrite, PASSTHROUGH } from "./pregel/write.js";
+import { TAG_HIDDEN } from "./constants.js";
+import { ManagedValueSpec } from "./managed/base.js";
+import { EphemeralValue } from "./channels/ephemeral_value.js";
+import { call, getRunnableForFunc } from "./pregel/call.js";
+import { RetryPolicy } from "./pregel/utils/index.js";
+import { Promisified } from "./pregel/types.js";
+
+/**
+ * Options for the @see task function
+ */
+export type TaskOptions = {
+  /**
+   * The retry policy for the task
+   */
+  retry?: RetryPolicy;
+};
+
+/**
+ * Wraps a function in a task that can be retried
+ *
+ *  !!! warning "Experimental"
```

Comment:
Do we actually want this warning in? Seems harsh

Reply:
Yea this is not the copy I'd want, I think better to say something like

!!! warning "Beta"
     This is a new API which might evolve over the next few weeks.
     We plan to mark it as stable by X day [next 3 weeks]

Reply:
I lifted it directly from the python lib. The `@experimental` and `@beta` tsdoc tags should ideally convey the same meaning, although per [the spec](https://tsdoc.org/pages/tags/beta/#suggested-meaning) they may be removed by tooling on non-beta releases. I haven't checked the docs build yet to see what actually happens, and whether we can configure this, so I left it in for now as a sort of reminder to myself.

Reply:
@nfcampos @eyurtsev is the current copy in the python lib what we're happy to go with here?

for reference:

```
    !!! warning "Beta"
        The Functional API is currently in beta and is subject to change.
```